README.md

```diff
@@ -1,32 +1,73 @@
-# Applicative option parser
-
-This package contains utilities and combinators to define command line option
-parsers.
+# optparse-applicative
 
 [![Continuous Integration status][status-png]][status]
 [![Hackage page (downloads and API reference)][hackage-png]][hackage]
 
+optparse-applicative is a haskell library for parsing options on the command line.
+
+Providing a powerful [applicative] interface to compose command line
+options and arguments, optparse-applicative provides all that is required for
+precise and concise CLI specifications for both simple and complex programs.
+
+optparse-applicative takes care of reading and validating the arguments passed
+to the command line, handling and reporting errors, generating a usage line, a
+comprehensive help screen, and enabling context-sensitive bash completions.
+
 **Table of Contents**
 
-- [Getting started](#getting-started)
-- [Supported options](#supported-options)
+- [Introduction](#introduction)
+- [Quick Start](#quick-start)
+- [Basics](#Basics)
+    - [Parsers](#parsers)
+    - [Applicative](#applicative)
+    - [Alternative](#alternative)
+    - [Running parsers](#running-parsers)
+- [Builders](#builders)
     - [Regular options](#regular-options)
     - [Flags](#flags)
     - [Arguments](#arguments)
     - [Commands](#commands)
-- [Option builders](#option-builders)
-- [Advanced features](#advanced-features)
+    - [Modifiers](#modifiers)
+- [Custom parsing and error handling](#custom-parsing-and-error-handling)
+    - [Parser runners](#parser-runners)
+    - [Option readers](#option-readers)
+    - [Preferences](#preferences)
+    - [Disambiguation](#disambiguation)
+    - [Customising the help screen](#customising-the-help-screen)
+    - [Command Groups](#command-groups)
+- [Bash completion](#bash-completion)
+    - [Actions and completers](#actions-and-completers)
+    - [Internals](#internals)
+- [Arrow interface](#arrow-interface)
+- [Applicative Do](#applicative-do)
+- [FAQ](#faq)
 - [How it works](#how-it-works)
 
-## Getting started
+## Introduction
 
-Install with
+The core type in optparse-applicative is a `Parser`
 
-```sh
-cabal install optparse-applicative
+```haskell
+data Parser a
+
+instance Functor Parser
+instance Applicative Parser
+instance Alternative Parser
 ```
 
-Here is a simple example of an applicative option parser:
+A value of type `Parser a` represents a specification for a set of options,
+which, if parsed correctly, will yield a value of type `a`.
```

Comment:
perhaps swap `, if parsed correctly,` like "which will yield a value of type `a` if parsing succeeds`